File: logisim/app/src/main/java/logisim/gui/main/AttrTableCircuitModel.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.gui.main;

import logisim.circuit.Circuit;
import logisim.circuit.CircuitMutation;
import logisim.data.Attribute;
import logisim.gui.generic.AttrTableSetException;
import logisim.gui.generic.AttributeSetTableModel;
import logisim.proj.Project;

public class AttrTableCircuitModel extends AttributeSetTableModel {
	private Project proj;
	private Circuit circ;

	public AttrTableCircuitModel(Project proj, Circuit circ) {
		super(circ.getStaticAttributes());
		this.proj = proj;
		this.circ = circ;
	}

	@Override
	public String getTitle() {
		return Strings.get("circuitAttrTitle", circ.getName());
	}

	@Override
	public void setValueRequested(Attribute<Object> attr, Object value) throws AttrTableSetException {
		if (!proj.getLogisimFile().contains(circ)) {
			String msg = Strings.get("cannotModifyCircuitError");
			throw new AttrTableSetException(msg);
		} else {
			CircuitMutation xn = new CircuitMutation(circ);
			xn.setForCircuit(attr, value);
			proj.doAction(xn.toAction(Strings.getter("changeCircuitAttrAction")));
		}
	}
}

File: LogisimPlus/LogisimPlus/logisim/gui/main/AttrTableCircuitModel.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.gui.main
{
	using Circuit = logisim.circuit.Circuit;
	using CircuitMutation = logisim.circuit.CircuitMutation;
	using logisim.data;
	using AttrTableSetException = logisim.gui.generic.AttrTableSetException;
	using AttributeSetTableModel = logisim.gui.generic.AttributeSetTableModel;
	using Project = logisim.proj.Project;

	public class AttrTableCircuitModel : AttributeSetTableModel
	{
		private Project proj;
		private Circuit circ;

		public AttrTableCircuitModel(Project proj, Circuit circ) : base(circ.StaticAttributes)
		{
			this.proj = proj;
			this.circ = circ;
		}

		public override string Title
		{
			get
			{
				return Strings.get("circuitAttrTitle", circ.Name);
			}
		}

// JAVA TO C# CONVERTER WARNING: Method 'throws' clauses are not available in C#:
// ORIGINAL LINE: @Override public void setValueRequested(logisim.data.Attribute attr, Object value) throws logisim.gui.generic.AttrTableSetException
		protected internal override void setValueRequested(Attribute attr, object value)
		{
			if (!proj.LogisimFile.contains(circ))
			{
				string msg = Strings.get("cannotModifyCircuitError");
				throw new AttrTableSetException(msg);
			}
			else
			{
				CircuitMutation xn = new CircuitMutation(circ);
				xn.setForCircuit(attr, value);
				proj.doAction(xn.toAction(Strings.getter("changeCircuitAttrAction")));
			}
		}
	}

}
